R/glue.R

```diff
@@ -287,12 +287,21 @@ trim <- function(x) {
 
 #' @export
 print.glue <- function(x, ..., sep = "\n") {
-  x[is.na(x)] <- style_na(x[is.na(x)])
-
-  if (length(x) > 0) {
-    cat(x, ..., sep = sep)
+  if (length(x) == 0) {
+    cat("<Length-0 glue vector>\n")
+    return(invisible(x))
   }
 
+  id <- format(paste0("[", seq_along(x), "] "), justify = "right")
+  indent <- paste0(style_grey(id, "|"), " ")
+  exdent <- paste0(strrep(" ", nchar(id[[1]])), style_grey("|"), " ")
+
+  out <- x
+  out[is.na(out)] <- style_red("NA")
+  out <- paste0(indent, out)
+  out <- gsub("\n", paste0("\n", exdent), out)
+  cat(paste0(out, "\n", collapse = ""))
+
```

Comment:
I think we should consider a max.print default like base R print methods have.

If you accidentally print `as_glue(1:1e6)` it takes around 10 seconds to print out for me.

But that will probably be finicky enough that it should be in a separate PR